### PR TITLE
Refactor Dotnet Dockerfiles

### PR DESCRIPTION
**Pull request by gtarique**
Follow this steps to build the docker images:
$ cd dotnet
$ docker build -t dotnetapp -f  Alpine\Dockerfile .

### CODE REVIEW

File: images/dotnet/Alpine/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+RUN chmod +x /app/entrypoint.sh
+COPY ../code/bin/Debug/net8.0/$APP_NAME.dll app/
+# Copy the published application into the image
+COPY --from=publish /app/publish .
+
+# Change the ownership of the application files to the new user and group
+RUN chown -R limitedaccessuser:limitedaccessuser /app
+
+# Switch to the non-root user
+USER limitedaccessuser
```
Comment (jacobbeasley) [inline images/dotnet/Alpine/Dockerfile]: needs to refer to user by id number

File: images/dotnet/Alpine/entrypoint.sh [hunk 2]
```diff
@@ -0,0 +1,2 @@
+#!/bin/sh
+dotnet ${APP_NAME}.dll
```
Comment (jacobbeasley) [inline images/dotnet/Alpine/entrypoint.sh]: May not need entrypoint.sh if its just one line like this. Could just change ENTRYPOINT to this: 

ENTRYPOINT dotnet ${APP_NAME}.dll

I think

Comment (gtarique) [reply images/dotnet/Alpine/entrypoint.sh]: updated as per your feedback

File: images/dotnet/Alpine/Dockerfile (hunk 1 above)
Comment (gtarique) [reply images/dotnet/Alpine/Dockerfile]: updated as per your feedback

**Comment by sahilnarang1** [comment]
> Follow this steps to build the docker images: $ cd dotnet $ docker build -t dotnetapp -f Alpine\Dockerfile .

@jacobbeasley I have tested. It is running with non root user. Also I have verified it is writing logs to stdout.

**Pull request merged**